core/clustersmngr/factory_caches.go

```diff
@@ -9,18 +9,52 @@ import (
 	"github.com/cheshir/ttlcache"
 	"github.com/weaveworks/weave-gitops/pkg/server/auth"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 type Clusters struct {
 	sync.RWMutex
 	clusters []Cluster
+	names    map[string]Cluster
 }
 
-func (c *Clusters) Set(clusters []Cluster) {
+func (c *Clusters) Set(clusters []Cluster) ([]Cluster, []Cluster) {
```

Comment:
We can godoc the exported `Set` function.

And, we can name the arguments for this to make it easier to understand...

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

> if a function returns two or three parameters of the same type, or if the meaning of a result isn't clear from context, adding names may be useful in some contexts.

it's returning two `[]Cluster` slices, we can name them `added` and `removed` which would help folks reading it (especially in Godoc) to understand what is returned?